### PR TITLE
CDAP-1223: New CLI command for Preferences to set preferences from a config file

### PR DESCRIPTION
**Pull request by gokulavasan**
JIRA : https://issues.cask.co/browse/CDAP-1223
Bamboo : https://builds.cask.co/browse/CDAP-RBT59

### CODE REVIEW

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/AbstractSetPreferencesCommand.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      case NAMESPACE:
+        if (programIdParts.length != 1) {
+          throw new CommandInputError(this);
+        }
+        client.setNamespacePreferences(programIdParts[0], args);
+        printSuccessMessage(printStream, type);
+        break;
+
+      case APP:
+        // todo : change this to 2 when namespace id is part of the argument.
```
Comment (jwang47) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/AbstractSetPreferencesCommand.java]: Actually I've added a `UseNamespace` command which allows you to set the current namespace in the CLI, and then you can retrieve the current namespace using `CLIConfig.getCurrentNamespace()`

**Comment by jwang47** [comment]
Just one comment.

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/AbstractSetPreferencesCommand.java (hunk 1 above)
Comment (gokulavasan) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/AbstractSetPreferencesCommand.java]: Oh nice. I will use it throughout Preferences CLI commands.

**Comment by gokulavasan** [comment]
@jwang47 Updated PR. Please take a look again. Thanks!

**Comment by jwang47** [comment]
LGTM

**Pull request merged**